common/kvstore/src/main/java/org/apache/spark/util/kvstore/KVStore.java

```diff
@@ -117,6 +118,11 @@
    */
   <T> KVStoreView<T> view(Class<T> type) throws Exception;
 
+  /**
+   * Returns a configurable view with condition for iterating over entities of the given type.
+   */
+  <T> KVStoreView<T> viewWithCondition(Class<T> type, Predicate<T> condition) throws Exception;
```

Comment:
I was going to ask, why not a `filter` method in `KVStoreView`, but that could be hard to get right given its logic around skip, first and last.